documentation/src/main/asciidoc/topics/code_examples/ProtoAdaptor.java

```diff
@@ -1,18 +1,23 @@
+import java.util.UUID;
+
+import org.infinispan.protostream.annotations.ProtoAdapter;
+import org.infinispan.protostream.annotations.ProtoFactory;
+import org.infinispan.protostream.annotations.ProtoField;
+import org.infinispan.protostream.descriptors.Type;
+
+/**
+ * Human readable UUID adapter for UUID marshalling
+ */
 @ProtoAdapter(UUID.class)
 public class UUIDAdapter {
 
-   @ProtoFactory
-   UUID create(Long mostSigBitsFixed, Long leastSigBitsFixed) {
-         return new UUID(mostSigBitsFixed, leastSigBitsFixed);
-   }
-
-   @ProtoField(number = 1, type = Type.FIXED64, defaultValue = "0")
-   Long getMostSigBitsFixed(UUID uuid) {
-      return uuid.getMostSignificantBits();
-   }
+  @ProtoFactory
+  UUID create(String stringUUID) {
+    return UUID.fromString(stringUUID);
+  }
 
-   @ProtoField(number = 2, type = Type.FIXED64, defaultValue = "0")
-   Long getLeastSigBitsFixed(UUID uuid) {
-      return uuid.getLeastSignificantBits();
-   }
+  @ProtoField(number = 1, required = true, type = Type.STRING)
```

Comment:
We shouldn't use `required`. Also no need to explicitly set the `type` attribute here, so this can be simplified to:

```java
@Protofield(1)
```